cmd/ui/src/views/Explore/ContextMenu/ContextMenu.tsx

```diff
@@ -18,77 +18,124 @@ import { Menu, MenuItem } from '@mui/material';
-                searchType,
-                primarySearch: selectedItemData?.objectId as string,
-            });
-        }
-    };
+const RX_EDGE_TYPE = /_(.*?)_/;
+
+const EdgeMenuItems: FC<MenuItemsProps> = ({ objectId, pathfindingFilters }) => {
+    const { handleRemoveEdgeType } = pathfindingFilters;
 
-    const handleSetEndingNode = () => {
-        const searchType = primarySearch ? 'pathfinding' : 'node';
-        const selectedItemData = selectedItemQuery.data;
-        if (selectedItemData && isNode(selectedItemData)) {
-            setExploreParams({
-                exploreSearchTab: 'pathfinding',
-                searchType,
-                secondarySearch: selectedItemData?.objectId as string,
-            });
+    const edgeType = objectId.match(RX_EDGE_TYPE)?.[1];
+
+    const filterEdge = () => {
+        if (edgeType) {
+            handleRemoveEdgeType(edgeType);
         }
     };
 
+    if (!edgeType) {
+        return null;
+    }
+
+    // Prevent filtering for edge types not found in AllEdgeTypes array
+    const item = isEdgeType(edgeType) ? (
+        <MenuItem key='filter-edge' onClick={filterEdge}>
+            Filter out Edge
+        </MenuItem>
+    ) : (
+        <MenuItem key='non-filterable' disabled>
+            Non-filterable Edge
+        </MenuItem>
+    );
+
+    return [item];
+};
+
+const NodeMenuItems: FC<Omit<MenuItemsProps, 'pathfindingFilters'>> = ({ objectId }) => {
+    const { checkPermission } = usePermissions();
+    const { primarySearch, secondarySearch, setExploreParams } = useExploreParams();
+    const { data: tierFlag } = useFeatureFlag('tier_management_engine');
+
+    const tierZeroId = useAppSelector(selectTierZeroAssetGroupId);
+    const ownedId = useAppSelector(selectOwnedAssetGroupId);
+
+    return [
+        <MenuItem
+            key='starting-node'
+            onClick={() =>
+                setExploreParams({
+                    exploreSearchTab: 'pathfinding',
+                    searchType: secondarySearch ? 'pathfinding' : 'node',
+                    primarySearch: objectId,
+                })
+            }>
+            Set as starting node
+        </MenuItem>,
+        <MenuItem
+            key='ending-node'
+            onClick={() =>
+                setExploreParams({
+                    exploreSearchTab: 'pathfinding',
+                    searchType: primarySearch ? 'pathfinding' : 'node',
+                    secondarySearch: objectId,
+                })
+            }>
+            Set as ending node
+        </MenuItem>,
+        ...(!tierFlag?.enabled && checkPermission(Permission.GRAPH_DB_WRITE)
+            ? [
+                  <AssetGroupMenuItem key='tier-zero' assetGroupId={tierZeroId} assetGroupName='High Value' />,
+                  <AssetGroupMenuItem key='owned' assetGroupId={ownedId} assetGroupName='Owned' />,
+              ]
+            : []),
+    ];
+};
+
+const ContextMenu: FC<{
+    contextMenu: Coordinates | null;
```

Comment:
Switched from the inline `{ mouseX, mouseY }` to the `Coordinates` type, which is already compatible with event mouse data.